### PR TITLE
[Jsonifier] New Port

### PR DESCRIPTION
**Pull request by RealTimeChris**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/jsonifier/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "jsonifier",
+  "version-date": "2023-03-18",
+  "description": "A few classes for parsing and serializing json - very rapidly.",
+  "homepage": "https://https://github.com/RealTimeChris/Jsonifier",
+  "license": "LGPL-2.1-or-later",
+  "supports": "(windows & x64 & !static) | (linux & x64)",
```
Comment (dg0yt) [inline ports/jsonifier/vcpkg.json:7]: Oh, this is tight. Why should handling JSON exclude arm or os or dynamic linkage?
Will have a look at upstream.

Comment (RealTimeChris) [reply ports/jsonifier/vcpkg.json:7]: I haven't written the fallback for arm - and don't you mean that I'm excluding static linkage? Tbh I forgot to check that individual part before pushing the repo I believe it does support static linkage.

**Comment by RealTimeChris** [comment]
Hey so it seems to be throwing a "read violation" in uwp upon attempting to run a necessary test to detect CPU architecture.
https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/86911/artifacts?artifactName=failure%20logs%20for%20x64-uwp&api-version=7.0&%24format=zip

**Comment by dg0yt** [comment]
uwp is a cross build. Simply do not run anything for the target, it is always a barrier for cross-builds.

**Comment by RealTimeChris** [comment]
> uwp is a cross build. Simply do not run anything for the target, it is always a barrier for cross-builds.

How do I not run anything for the target?

**Comment by dg0yt** [comment]
> > uwp is a cross build. Simply do not run anything for the target, it is always a barrier for cross-builds.
> 
> How do I not run anything for the target?

Do not use any `*_SOURCE_RUNS` or similar (here: `FindAVX.cmake`).
Or make the port directly set the expected results for each target CPU, so that the actual test is skipped.

File: ports/jsonifier/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "jsonifier",
+  "version": "0.9.0",
+  "description": "A few classes for parsing and serializing json - very rapidly.",
+  "homepage": "https://https://github.com/RealTimeChris/Jsonifier",
+  "license": "LGPL-2.1-or-later",
+  "supports": "(windows & x64 & !uwp) | (linux & x64) | (osx & x64)",
```
Comment (jimwang118) [inline ports/jsonifier/vcpkg.json:7]: Doesn't this library upstream support x86 builds?

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/jsonifier/vcpkg.json (hunk 2 above)
Comment (RealTimeChris) [reply ports/jsonifier/vcpkg.json:7]: No I don't believe it does as it utilizes AVX/AVX2/AVX512 instruction sets.

File: ports/jsonifier/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+file(
+	INSTALL "${SOURCE_PATH}/License.md"
+	DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
+	RENAME copyright
+)
```
Comment (jimwang118) [inline ports/jsonifier/portfile.cmake:21]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/License.md")
```

File: ports/jsonifier/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "jsonifier",
+  "version": "0.9.0",
+  "description": "A few classes for parsing and serializing json - very rapidly.",
+  "homepage": "https://https://github.com/RealTimeChris/Jsonifier",
```
Comment (jimwang118) [inline ports/jsonifier/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/RealTimeChris/Jsonifier",
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by dg0yt** [comment]
I still wonder whether this lib requires C++20 or C++23.

Being header-only, it really shouldn't deal with AVX detection at port build time IMO. Artifact caching will easily give the wrong package. If not autodetected at runtime, a possible solution is to create independet targets (as in ports fbgemm, libsodium).

**Comment by RealTimeChris** [comment]
> I still wonder whether this lib requires C++20 or C++23.
> 
> Being header-only, it really shouldn't deal with AVX detection at port build time IMO. Artifact caching will easily give the wrong package. If not autodetected at runtime, a possible solution is to create independet targets (as in ports fbgemm, libsodium).

It uses std::expected, so it requires C++23. Also that's an interesting idea RE: Independent Targets.

File: ports/jsonifier/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/License.md")
```
Comment (jimwang118) [inline ports/jsonifier/portfile.cmake:17]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/License.md")

```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Tested usage successfully by the x64-windows triplet.
```
find_package(Jsonifier CONFIG REQUIRED)
target_link_libraries(main PRIVATE Jsonifier::Jsonifier)
```

**Review by jimwang118 — APPROVED** [review]

**Pull request merged**